ls1intum/Artemis
# `Programming exercises`: Fix missing test names during manual assessment

### Checklist
#### General
#### Server
#### Client

### Motivation and Context
My PR #7015 introduced a small bug where tutors would not see test case names during assessment. Fixes #7554.

### Description
- This PR fixes the issue and passes test cases to the tutor assessment.
- Fixed an issue when saving manual programming assessments.
- Reduced the transferred data when loading long feedback texts. 

### Steps for Testing
Prerequisites:
- 1 Programming Exercise with test cases generating long feedback (e.g. by calling `fail("a".repeat(2000));`) and manual assessment enabled

1. As a tutor, start a new assessment
2. Check that test names are shown in the feedback detail view.
3. Check that loading long feedback text still works.
4. Assess the submission and save it.
5. Check that viewing test case names still works afterward (for tutors and students).

#### Exam Mode Testing
same as above, but for an exam assessment.

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress
#### Performance Review
#### Code Review
#### Manual Tests
#### Exam Mode Test
